ptvsd/wrapper.py

```diff
@@ -127,28 +128,10 @@ def using_format(self, fmt):
 str_handlers = pydevd_extutil.EXTENSION_MANAGER_INSTANCE.type_to_instance.setdefault(pydevd_extapi.StrPresentationProvider, [])  # noqa
 str_handlers.insert(0, SafeReprPresentationProvider._instance)
 
-DONT_TRACE_FILES = set((os.path.join(*elem) for elem in [
-    ('ptvsd', 'attach_server.py'),
-    ('ptvsd', 'daemon.py'),
-    ('ptvsd', 'debugger.py'),
-    ('ptvsd', 'futures.py'),
-    ('ptvsd', 'ipcjson.py'),
-    ('ptvsd', 'pathutils.py'),
-    ('ptvsd', 'pydevd_hooks.py'),
-    ('ptvsd', 'reraise.py'),
-    ('ptvsd', 'reraise2.py'),
-    ('ptvsd', 'reraise3.py'),
-    ('ptvsd', 'runner.py'),
-    ('ptvsd', 'safe_repr.py'),
-    ('ptvsd', 'socket.py'),
-    ('ptvsd', 'untangle.py'),
-    ('ptvsd', 'version.py'),
-    ('ptvsd', 'wrapper.py'),
-    ('ptvsd', '_main.py'),
-    ('ptvsd', '_version.py'),
-    ('ptvsd', '__init__.py'),
-    ('ptvsd', '__main__.py'),
-]))
+PTVSD_DIR_PATH = os.path.dirname(__file__)
+DONT_TRACE_FILES = set((os.path.join(*elem) for elem in
+                        (('ptvsd', f) for f in os.listdir(PTVSD_DIR_PATH)
+                        if f.endswith('.py'))))
```

Comment:
This is far less robust than you think. Better to do `import ptvsd` and then exclude any frame where its filename is within any of the elements of `ptvsd.__path__` (accounting for `normcase`, of course).

The aim is to avoid hitting the file system at all, and make no assumptions about whether the files actually exist on disk (or whether they are `.py` and not `.pyc` etc.)

Reply:
I used this list to tell pydevd to skip tracing on these files, in the frame tracing call back. See https://github.com/Microsoft/ptvsd/blob/master/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_frame.py#L730
Are you suggesting we do this manually be removing frames when we suspend threads?

Reply:
No, just suggesting changing the comparison function to do a `.startswith()` instead of `in` comparison.